src/lib/blocks.js

```diff
@@ -110,6 +110,75 @@ export default function (vm) {
         return [[myself, '_myself_']].concat(spriteMenu());
     };
 
+    const variablePropertyMenu = function (thisValue) {
+        // Fill the output array with sprite properties by default. This will be changed to stage properties if the
+        // stage is selected.
+        const xPosition = ScratchBlocks.ScratchMsgs.translate('SENSING_OF_XPOSITION', 'x position');
+        const yPosition = ScratchBlocks.ScratchMsgs.translate('SENSING_OF_YPOSITION', 'y position');
+        const direction = ScratchBlocks.ScratchMsgs.translate('SENSING_OF_DIRECTION', 'direction');
+        const costumeNumber = ScratchBlocks.ScratchMsgs.translate('SENSING_OF_COSTUMENUMBER', 'costume #');
+        const costumeName = ScratchBlocks.ScratchMsgs.translate('SENSING_OF_COSTUMENAME', 'costume name');
+        const size = ScratchBlocks.ScratchMsgs.translate('SENSING_OF_SIZE', 'size');
+        const volume = ScratchBlocks.ScratchMsgs.translate('SENSING_OF_VOLUME', 'volume');
+        let output = [
+            [xPosition, 'x position'],
+            [yPosition, 'y position'],
+            [direction, 'direction'],
+            [costumeNumber, 'costume #'],
+            [costumeName, 'costume name'],
+            [size, 'size'],
+            [volume, 'volume']
+        ];
```

Comment:
The output array is initially created with these values so that if, for any reason, neither a sprite nor the stage can be found based on the object input, the sprite-specific options will be the ones which are shown by default.